File: app/src/main/java/com/sunny/family/layout/fragment/XxxCoordinatorFragment.java
package com.sunny.family.layout.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;

import com.sunny.family.R;
import com.sunny.lib.base.BaseFragment;

/**
 * Created by zhangxin17 on 2020/11/19
 */
public class XxxCoordinatorFragment extends BaseFragment {

    private View rootLayout;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        rootLayout = inflater.inflate(R.layout.fragment_home_base, container, false);
        AppCompatTextView nameView = rootLayout.findViewById(R.id.fragment_name);
        nameView.setText("协调布局");
        return rootLayout;
    }

}

File: moduleView/src/main/java/com/sunny/module/view/xxx/coordinator/XxxCoordinatorFragment.kt
package com.sunny.module.view.xxx.coordinator

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.sunny.lib.base.log.SunLog
import com.sunny.lib.common.base.BaseFragment
import com.sunny.lib.common.city.CityAdapter
import com.sunny.lib.common.city.CityItemModel
import com.sunny.lib.common.city.buildCityData
import com.sunny.module.view.R
import com.sunny.module.view.layout.CoordinatorLayoutActivity

/**
 * Created by zhangxin17 on 2020/11/19
 */
class XxxCoordinatorFragment : BaseFragment() {

    lateinit var mRecyclerView: RecyclerView
    lateinit var mAdapter: CityAdapter

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.view_fragment_xxx_coordinator, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView(view)
    }

    fun initView(root: View) {
        mRecyclerView = root.findViewById(R.id.video_recycler_view)
        initRecyclerView()
    }

    private fun initRecyclerView() {
        mRecyclerView.layoutManager = GridLayoutManager(requireContext(), 1)
        mRecyclerView.setHasFixedSize(true)

        mAdapter = CityAdapter(buildCityData())
        mRecyclerView.adapter = mAdapter

        mAdapter.setOnItemClickListener { adapter, view, position ->

            val itemModel: CityItemModel = adapter.getItem(position) as CityItemModel
            if (itemModel.isCity()) {
                showToast(itemModel.showName)
            }
        }
        mRecyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {

            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                super.onScrolled(recyclerView, dx, dy)

                val stickView = recyclerView.findChildViewUnder(0f, 0f)

                SunLog.i(CoordinatorLayoutActivity.TAG, "onScrolled $stickView")
            }
        })
    }
}